### ISSUE TITLE
Avoid sending ContentType with GET requests

### ISSUE THREAD

**Issue by niicojs** [alex-cory/use-http#133]
**Describe the bug**
useFetch sends a content type header (application/json by default) with GET requests. This should not happend and is making express fail by default

**To Reproduce**
Make a GET request without any options, use-http sends a content type header of "application/json"

**Expected behavior**
ContentType header should not be present when making get requests

**Comment by niicojs** [comment]
Workaround in the meantime:
   ```
 interceptors: {
      request: options => {
        if (options && options.headers) {
          delete options.headers['Content-Type'];
        }
        return options;
      }
    }
```

**Comment by alex-cory** [comment]
Your solution works. Actually, this default was based off of [axios](https://github.com/axios/axios/blob/1b07fb9365d38a1a8ce7427130bf9db8101daf09/lib/defaults.js)'s default on this. If you don't want `applicatoin/json` as your content-type for all requests, your solution is the correct one. If you are doing something with `FormData` which so far is the main reason why you would not want this header, then when you POST, as the body use an instance of FormData. If you just want to remove this for one, request, you can do
```js
const [request] = useFetch(globalOptions => ({
  delete globalOptions.headers['Content-Type']
  return globalOptions
}))
```

**Comment by niicojs** [comment]
The issue is that you are currently sending a content type with get requests which is invalid according to http specs.
My solution works but I shouldn't have to do this.

**Comment by alex-cory** [comment]
Okay, I just looked at axios again. Looking at the request headers from the test, I didn't see any content-type header.  I think I'll go ahead and remove it.

**Comment by alex-cory** [comment]
This should be fixed. See `v0.1.96`. If the issue persists, let me know and I'll have another look.

**Comment by niicojs** [comment]
Doesn't seem to work. 
May be related: https://github.com/alex-cory/use-http/blob/21bfece642d865e71fb2b22d209818c30bdd176f/src/makeRouteAndOptions.ts#L54

**Comment by alex-cory** [comment]
This should be fixed now. Apologies I had the `Content-Type` in 2 different places and didn't realize. Should be fixed now in `v0.1.98`

**Comment by niicojs** [comment]
Thanks. This fix the original issue of GET requests. However, not sending the json content type when sending other types of requests (like PUT and POST) is breaking existing code.
Maybe remove default content type only for GET?

**Comment by alex-cory** [comment]
I'm only going to **default** the `Content-Type: application/json` for POST and PUT requests. This decision is due to [this stackoverflow question](http://bit.ly/2NJNt3N). This PR should fix the remaining issues here. https://github.com/alex-cory/use-http/pull/141

**Comment by alex-cory** [comment]
This should be published in `v0.1.99`

**Comment by niicojs** [comment]
Thanks man, works fine.
